numpy/ma/core.py

```diff
@@ -5869,6 +5869,18 @@ def __getitem__(self, indx):
 
         """
         m = self._mask
+        if m[indx].size > 1:
```

Comment:
It appears that crazier dtypes do not get assigned to the mask properly anyway:

```
In [234]: Z = ma.zeros(2, dtype([("A", "(2,2)i1,(2,2)i1", (2,2))]))

In [237]: print(Z.dtype.descr)
[('A', [('f0', '|i1', (2, 2)), ('f1', '|i1', (2, 2))], (2, 2))]

In [238]: print(Z.mask.dtype.descr)
[('A', '|b1', (2, 2))]
```

That's bound to lead to problems somewhere as well...